### PR TITLE
#168781627 - Users can view all requests and a single request

### PR DESCRIPTION
**Pull request by nignanthomas**
#### What does this PR do?
Users can view all requests and a single request

#### Description of Task to be completed?
- Create menuBar component
- Create ViewRequests component
- Create requestItem component
- Create ViewSingleRequest component
- Create tests

#### How should this be manually tested?
- Clone ft-view-requests-168781627 branch
- run `npm i`
- run `npm run dev`
- Click on `Requests` in the MenuBar

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#168781627](https://www.pivotaltracker.com/story/show/168781627)

#### Screenshots (if appropriate)
Screen Shot 2019-12-11 at 15 18 06
Screen Shot 2019-12-12 at 09 49 49
Screen Shot 2019-12-12 at 09 50 36

#### Questions:
N/A

### CODE REVIEW

**Review by ngireric123 — APPROVED** [review]
LGTM

File: src/tests/mocks/requestsMocks.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    reasons: 'Partner Company Meeting',
+    arrivalDate: '2019-11-21',
+    departureDate: null,
+    location: {
+      country: 'Nigeria',
+      id: 5,
+      name: 'Lagos Office',
+    },
+  },
+
```
Comment (sabin18) [inline src/tests/mocks/requestsMocks.js:97]: remove this line

File: src/tests/requestsTests/SingleRequestTest.spec.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    return wrapper;
+} 
+
+describe('SingleRequest Test Suite', () => { 
+  it('Should Mount Successfully', () => {
+    const component = setUp(mainState); 
+    expect(component.find(MenuComponent)).toHaveLength(1);
+  });
+
+  // it('Should Simulate Successfull Signup', () => {
```
Comment (sabin18) [inline src/tests/requestsTests/SingleRequestTest.spec.js]: remove comments here.

File: src/tests/requestsTests/ViewRequests.spec.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    return wrapper;
+} 
+
+describe('ViewRequests Test Suite', () => { 
+  it('Should Mount Successfully', () => {
+    const component = setUp(mainState); 
+    expect(component.find(MenuComponent)).toHaveLength(1);
+  });
+
+  // it('Should Simulate Successfull Signup', () => {
```
Comment (sabin18) [inline src/tests/requestsTests/ViewRequests.spec.js]: remove commented lines here.

File: src/styles/styles.scss [hunk 4]
```diff
@@ -1,10 +1,17 @@
...
+@import './requests/index.scss';
+@import './breadcrumbs.scss';
+@import './menuBar.scss';
+@import '~bootstrap/scss/bootstrap';
+
 * {
 	font-family: $master-font;
 	margin: 0;
 	padding: 0;
+	// border-radius: 0 !important;
```
Comment (sabin18) [inline src/styles/styles.scss]: remove this commented line.

File: src/styles/requests/index.scss [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+@import './node_modules/react-day-picker/lib/style';
+
+
+.row {
+	display: flex;
+	flex-direction: row;
+	width: 80vw;
+	padding: 15px;
+}
+// .row-end {
```
Comment (sabin18) [inline src/styles/requests/index.scss]: remove these commented lines.

**Review by sabin18 — CHANGES_REQUESTED** [review]
Good job @nignanthomas, work on comments I left.

**Review by sabin18 — APPROVED** [review]

File: src/styles/menuBar.scss [hunk 6]
```diff
@@ -0,0 +1,39 @@
+.account-icon {
+	display: flex;
+	flex-direction: column;
+	align-items: center;
+	font-size: 12px;
+	justify-content: space-around;
+	text-transform: uppercase;
+	color: $white;
+	div {
+		display: flex;
```
Comment (nakiwuge) [inline src/styles/menuBar.scss]: Indent this

File: src/styles/menuBar.scss [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+	display: flex;
+	flex-direction: row;
+	justify-content: space-evenly;
+	align-items: center;
+	text-transform: uppercase;
+	letter-spacing: 2px;
+	font-size: smaller;
+	img {
+		height: 35px;
+		width: 35px;
```
Comment (nakiwuge) [inline src/styles/menuBar.scss]: Indent this

File: src/styles/requests/destinations.scss [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+.destination-item {
+	margin: 10px 0;
+	display: flex;
+	flex-direction: row;
+	margin: 0px 15px;
+	width: 100%;
+	flex-wrap: wrap;
+
+	.pad-items {
+		display: flex;
```
Comment (nakiwuge) [inline src/styles/requests/destinations.scss]: Indent this

File: src/styles/requests/index.scss [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+
+.form-section {
+	display: flex;
+	justify-content: space-evenly;
+	width: 100vw;
+	flex-wrap: wrap;
+
+	.select {
+		width: 20vw;
+		display: flex;
```
Comment (nakiwuge) [inline src/styles/requests/index.scss]: Indent this

File: src/views/requests/SingleRequest.js [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+} from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import DayPickerInput from 'react-day-picker/DayPickerInput';
+import { singleRequestAction } from '../../actions/requestsActions';
+import Breadcrumbs from '../../components/global/Breadcrumbs';
+import MenuComponent from '../../components/global/MenuComponent';
+import Alert from '../../components/global/AlertComponent';
+import DestinationDisplay from '../../components/pages/requests/DestinationDisplay';
+
+
```
Comment (nakiwuge) [inline src/views/requests/SingleRequest.js]: remove this line

File: src/views/requests/ViewRequests.js [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+} from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import { Add } from '@material-ui/icons';
+import { getRequestsAction } from '../../actions/requestsActions';
+import Breadcrumbs from '../../components/global/Breadcrumbs';
+import MenuComponent from '../../components/global/MenuComponent';
+import RequestItem from '../../components/pages/requests/RequestItem';
+import Alert from '../../components/global/AlertComponent';
+
+
```
Comment (nakiwuge) [inline src/views/requests/ViewRequests.js]: remove this line

**Review by nakiwuge — CHANGES_REQUESTED** [review]
@nignanthomas good job on the implementation however you need to create a `protectedRoute` component for the routes you don't want a user to access without logging in. You also need to check if the token provided is valid or not expired. If the token is invalid/expired redirect the user to the login page.
Also, it is bad practice to import the navbar in every component. 
Checkout these links to get started
- https://medium.com/@leonardobrunolima/react-tips-how-to-protect-routes-for-unauthorized-access-with-react-router-v4-73c0d451e0a2
- https://dev.to/calier/setting-up-an-authenticated-route-in-react-4m9a
- https://www.youtube.com/watch?v=Y0-qdp-XBJg

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**